### PR TITLE
[REF] refactor permute_ols

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- create functions:
  - _compute_t_stat_threshold function
  - _make_array_contiguous
- return early if not permutation needed
- only compute things needed for permutation
- ensures that n_perm is an int

### CODE REVIEW

File: nilearn/mass_univariate/permuted_least_squares.py [hunk 1]
```diff
@@ -948,7 +935,36 @@ def permuted_ols(
...
+def _make_array_contiguous(array):
+    """Make arrays contiguous for code efficiency."""
+    if not array.flags["C_CONTIGUOUS"]:
+        # useful to developer
+        warnings.warn("Target variates not C_CONTIGUOUS.")
+        array = np.ascontiguousarray(array)
+    return array
+
+
+def _compute_t_stat_threshold(
```
Comment (bthirion) [inline nilearn/mass_univariate/permuted_least_squares.py:947]: one-line docstrings are always welcome for this type of functions

**Review by bthirion — COMMENTED** [review]
LGTM so far.

File: nilearn/mass_univariate/permuted_least_squares.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/mass_univariate/permuted_least_squares.py:947]: done 
and replaced dof by degrees_of_freedom

File: nilearn/mass_univariate/permuted_least_squares.py [hunk 2]
```diff
@@ -948,7 +935,37 @@ def permuted_ols(
     return outputs
 
 
-def _check_inputs_permuted_ols(n_jobs, tfce, masker, threshold, target_vars):
+def _make_array_contiguous(array):
+    """Make arrays contiguous for code efficiency."""
+    if not array.flags["C_CONTIGUOUS"]:
+        # useful to developer
+        warnings.warn("Target variates not C_CONTIGUOUS.")
```
Comment (man-shu) [inline nilearn/mass_univariate/permuted_least_squares.py:942]: Codecov is complaining about testing this warning

File: nilearn/mass_univariate/permuted_least_squares.py [hunk 3]
```diff
@@ -948,7 +935,37 @@ def permuted_ols(
...
+        array = np.ascontiguousarray(array)
+    return array
+
+
+def _compute_t_stat_threshold(
+    threshold, two_sided_test, tested_vars, confounding_vars
+):
+    """Compute t-stat threshold if needed based degrees of freedom."""
```
Comment (man-shu) [inline nilearn/mass_univariate/permuted_least_squares.py:950]:
```suggestion
    """Compute t-stat threshold if needed based on degrees of freedom."""
```

**Review by man-shu — APPROVED** [review]
Couldn't do a deeper review because I am not very sure what's going on algorithm-wise.
But here are a couple superficial comments...

File: nilearn/mass_univariate/permuted_least_squares.py (hunk 2 above)
Comment (Remi-Gau) [reply nilearn/mass_univariate/permuted_least_squares.py:942]: Yeah I don't think I will bother with this one for now

**Comment by Remi-Gau** [comment]
linkcheck failure is unrelated: wanna see if we get it systematically

though in general I'd rather just ignore flacky links

**Pull request merged**